stdlib/SparseArrays/src/higherorderfns.jl

```diff
@@ -1154,4 +1155,23 @@ map(f::Tf, A::SparseOrStructuredMatrix, Bs::Vararg{SparseOrStructuredMatrix,N})
 map!(f::Tf, C::AbstractSparseMatrixCSC, A::SparseOrStructuredMatrix, Bs::Vararg{SparseOrStructuredMatrix,N}) where {Tf,N} =
     (_checksameshape(C, A, Bs...); _noshapecheck_map!(f, C, _sparsifystructured(A), map(_sparsifystructured, Bs)...))
 
+
+# (13) extrema methods optimized for sparse vectors/matrices.
+function extrema(f, A::SparseVecOrMat)
+    if iszero(length(A))
+        throw(ArgumentError("Sparse array must have at least one element."))
+    end
+    N = nnz(A)
+    iszero(N) && return f(zero(eltype(A))), f(zero(eltype(A)))
+    vmin, vmax = extrema(f, nonzeros(A))
+    if N != length(A)
+        f0 = f(zero(eltype(A)))
+        vmin = min(f0, vmin)
+        vmax = max(f0, vmax)
+    end
+    vmin, vmax
+end
+
+extrema(A::SparseVecOrMat) = extrema(identity, A)
```

Comment:
I believe this fallback is already defined in Base.

Reply:
The more specific definition is
`extrema(A::AbstractArray; dims = :) = _extrema_dims(identity, A, dims)`
in base/multidimensional.jl line 1599.
So it seems this is needed.
I benchmarked `extrema` with:
```
julia> S = sprand(10000, 10000, 0.00001);
julia> @btime extrema(S)
  3.328 μs (1 allocation: 32 bytes)
(0.0, 0.9993973695872556)
```
and without this line:
```
julia> @btime extrema(S)
  643.939 ms (1 allocation: 32 bytes)
(0.0, 0.9993973695872556)
```